test/pub_sub.coffee

```diff
@@ -17,17 +17,15 @@ describe 'PubSub', ->
     deferred.promise
 
   describe '#subscribe', ->
-    it 'should subscribe to a pubsub channel', (done) ->
+    it 'should subscribe to a pubsub channel', ->
```

Comment:
Be careful when removing `done` because that can give you evergreen tests for async calls. See https://wietse.loves.engineering/testing-promises-with-mocha-90df8b7d2e35

Reply:
thanks, i was hesitant to remove the done calls. The issue is with Coffeescript and the implicit return operator clashes with done when using promises, https://mochajs.org/#working-with-promises

> In Mocha v3.0.0 and newer, returning a Promise and calling done() will result in an exception, as this is generally a mistake:

Reading suggests returning the promise is fine as mocha was ignoring the done() call before as it was. 

https://mochajs.org/#working-with-promises
> Alternately, instead of using the done() callback, you may return a Promise. This is useful if the APIs you are testing return promises instead of taking callbacks:

Advice here is appreciated.

Reply:
Returning the Promise sounds like it should work. After reading that post, I’ve been terminating my tests with a second then block: `.then(done, done)`.  The second `done` makes sure your test terminates on failure too.

Reply:
This came up in a recent issue on the monorepo where a test for `wrapper.find(X)` was passing but it turned out the component didn’t render X. I think it’s caused by mocha throwing errors to indicate failure, and the expectation is evaluated before the asynchronous function can generate the error. I’m not sure.